syntax = "proto2";

package caffe;

  ...
  ...
  ...
  
// NOTE
// Update the next available ID when you add a new LayerParameter field.
//
// LayerParameter next available layer-specific ID: 148 (last added: elementwise_product_param)
message LayerParameter {
	optional string name = 1; // the layer name
	optional string type = 2; // the layer type
	repeated string bottom = 3; // the name of each bottom blob
	repeated string top = 4; // the name of each top blob

	...
	...
	...
	optional RecurrentParameter recurrent_param = 146;
	optional ElementWiseProductParameter elementwise_product_param = 147;
}

  ...
  ...
  ...

message ElementWiseProductParameter {
	optional bool bias_term = 1 [default = false]; // whether to have bias terms
	optional FillerParameter weight_filler = 2 ; // The filler for the weight
	optional FillerParameter bias_filler = 3 ; // The filler for the bias
	optional int32 share = 4 [default = 1] ; // Share weights with x channels
}